Add osint-web-check extension

gksander opened the pull request:
## Description

This PR adds a `osint-web-check` extension (Open Source Intelligence) motivated by [this repo](https://github.com/lissy93/web-check) that takes a URL and runs various OSINT checks against the URL (DNS, SSL, DNSSEC, HTTP Headers, etc.).

## Screencast

| | |
| --- | --- |
| Input | Example (WHOIS) |
| <img src="https://github.com/raycast/extensions/assets/12721310/565a698d-6b67-40a8-8171-1bd96aa1a47a" /> | <img src="https://github.com/raycast/extensions/assets/12721310/dc8fbb3b-eb02-4d12-8fd8-2d1f7bcb2d95" />  |
| Example (robots.txt) | Info view |
| <img src="https://github.com/raycast/extensions/assets/12721310/3cfd7cfd-1f12-436c-be19-19cfea5efa34" /> | <img src="https://github.com/raycast/extensions/assets/12721310/6ea5c18d-8708-4172-9b88-8fa6333c3134" /> |

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- You can add `tintColor` to your icons like above, that could maybe give some extra attention?
![osint-web-check 2024-01-09 at 12 06 25](https://github.com/raycast/extensions/assets/38033107/826ad077-80ef-4006-9df8-0e1fa44cf0c3)

- You could use navigationTitle to append the domain, you're allowed to do that since you push to a new view 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

gksander commented:
Thank you for the feedback @pernielsentikaer! Added in your suggestions with `tintColor` and `navigationTitle`, looks much better 🎉 

<img width="862" alt="image" src="https://github.com/raycast/extensions/assets/12721310/b62ac45a-0477-4e21-8020-5798b5c0356d">

pernielsentikaer commented:
You could also add an argument so you don't need the form maybe? Something like this

![Raycast09012024-FVyo05UE png@2x](https://github.com/raycast/extensions/assets/38033107/8ac2e254-690b-4b6b-99fd-5d1574b055e8)

Not sure if you like that?

gksander commented:
Oh neat. Is there any way to do validation with arguments? Right now with the form, I'm doing a DNS `lookup` to ensure the input is a valid url to do a check against – wondering if there's any way to do custom validation with arguments.

pernielsentikaer commented:
No that's not possible, but it would be cool - You could use the argument and only show the form if the validation fails 🙂

pernielsentikaer commented:
Might be a bit too much for v1?

gksander commented:
Oh nice, I like that idea! Updated to do just that.

https://github.com/raycast/extensions/assets/12721310/0543d25c-6424-43e1-a1a9-accb2ee118e1

pernielsentikaer commented on extensions/osint-web-check/src/osint-web-check.tsx:
```diff
@@ -0,0 +1,72 @@
+
+  return (
+    <UrlInput
+      onSubmit={(url) => {
```
```suggestion
      url={url}
      onSubmit={(url) => {
```

pernielsentikaer commented on extensions/osint-web-check/src/UrlInput.tsx:
```diff
@@ -0,0 +1,49 @@
+
+type UrlInputValues = { url: string };
+
+type UrlInputProps = {
+  onSubmit: (url: string) => void;
+};
```
```suggestion
type UrlInputProps = {
  onSubmit: (url: string) => void;
  url: string;
};
```

pernielsentikaer commented on extensions/osint-web-check/src/UrlInput.tsx:
```diff
@@ -0,0 +1,49 @@
+};
+
+export function UrlInput({ onSubmit }: UrlInputProps) {
+  const [urlError, setUrlError] = useState<string | undefined>(undefined);
```
```suggestion
  const [urlError, setUrlError] = useState<string | undefined>("Enter a valid http(s) URL.");
```

pernielsentikaer commented on extensions/osint-web-check/src/UrlInput.tsx:
```diff
@@ -0,0 +1,49 @@
+      <Form.TextField
+        id="url"
+        title="URL"
+        placeholder="google.com"
```
```suggestion
        placeholder={url}
```

gksander replied:
In this case, `url` will be a *bad* url, so we wouldn't want that as a placeholder. I'm instead using it as an initial value, and opting for a controlled form here – so upon entry here, we'll show the "bad" url they just entered, along with the error message/state.

pernielsentikaer reviewed (changes requested):
That's a really awesome update 🔥 

Do you think something like suggested (or check `useForm`) would make sense to add?

gksander commented:
Okay, took the heart of your suggestions and incorporated them in! Of note, changed to a controlled form for `UrlInput` so we can pass the "bad" url they initially entered in as an initial value and have an error state.

https://github.com/raycast/extensions/assets/12721310/92781aee-97d1-4060-b5a5-bbb75e6accb0

gksander commented:
Think I'm going to push up one more commit in a little bit to get rid of the loading Detail view when the initial DNS lookup is happening. I think the List view can just pause for a short bit while we do the initial validation, feels like it'd be less jank and a smoother UX.

gksander commented:
Okay, pushed up another commit that cleans up a lot of the repeated code, and assumes user will take "the happy path" and enter a valid URL, so we don't need an intermediate step. If user enters invalid url, we'll then kick them back to the manual form with the error notice.

https://github.com/raycast/extensions/assets/12721310/9bf60905-1610-493e-8ac1-4fefa5686ea3

pernielsentikaer commented:
What a huge difference it makes! I really love the flow now. Awesome work! 🚀

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
